include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,268 @@
+  // Internal concept checking if an option is supported by a any specification in a given subset
+  //====================================================================================================================
+  template<typename O, typename... Specs>
+  concept is_supported = (supports_option<Specs,O> || ... );
+
+  //====================================================================================================================
+  // Internal option carrying conditional mask or conditional expressions
+  //====================================================================================================================
+  struct condition_ : rbr::as_keyword<condition_>
+  {
+    using rbr::as_keyword<condition_>::operator=;
+    template<typename T> static constexpr bool check()
+    {
+      return conditional_expr<std::remove_cvref_t<T>> || logical_value<std::remove_cvref_t<T>>;
+    }
+  };
+}
+
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @var condition
+  //!   @brief Keyword for retrieving conditionals decorator
+  //!
+  //!   When a eve::decorated_callable is called with an option provided via eve::conditional, the value of the
+  //!   conditional can be retrieved using eve::condition
+  //!   @see eve::supports
+  //! @}
+  //====================================================================================================================
+  inline constexpr detail::condition_ condition = {};
+}
+
+namespace eve::detail
+{
+  //====================================================================================================================
+  // Intermediate type for handling chains of option like in func[opt1 = y][opt2 = x][ignore](...)
+  //====================================================================================================================
+  template<typename Tag, typename Settings, typename... Specs>
+  struct decorated_fn : Specs...
+  {
+    protected:
+    using Specs::process_option...;
+
+    public:
+    using callable_tag_type = void;
+
+    decorated_fn(Settings s) : opts(std::move(s)) {}
+
+    template<typename O>
+    EVE_FORCEINLINE auto operator[](O const& o) const
+    -> decltype ( decorated_fn<Tag,decltype(process_option(std::declval<Settings>(), o)),Specs...>
+                  {process_option(std::declval<Settings>(), o)}
+                )
+    {
+      auto new_opts = process_option(opts, o);
+      return decorated_fn<Tag,decltype(new_opts),Specs...>{new_opts};
+    }
+
+    // Behave as a callable but pass the current bundle of options to the deferred_call handler
+    template<typename... Args>
+    EVE_FORCEINLINE constexpr auto operator()(Args&&... args) const
+    -> decltype(std::declval<Tag>().call(std::declval<Args>()...))
+    {
+      return Tag::deferred_call(eve::current_api, opts, EVE_FWD(args)...);
```

Comment:
Where is defaults?

Reply:
in protocol.hpp

```c++
#define EVE_CALLABLE_OBJECT_FROM(NS,TYPE,NAME)                                                                         \
template<typename... Args>                                                                                             \
EVE_FORCEINLINE constexpr auto operator()(Args&&... args) const                                                        \
-> decltype(std::declval<TYPE>().call(std::declval<Args>()...))                                                        \
requires( requires { std::declval<TYPE>().call(EVE_FWD(args)...); })                                                   \
{                                                                                                                      \
  return TYPE::deferred_call(eve::current_api, eve::detail::defaults<TYPE,Args...>(), EVE_FWD(args)...);               \
}                                                                                                                      \
```

Where do you want it to be ?

Reply:
Here - you populate options here - fil in the defaults too. I think we don't yet agree on what defaults are